src/utils/latency_tracer.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2017-present, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+#include "latency_tracer.h"
+
+#include <dsn/service_api_c.h>
+#include <dsn/dist/fmt_logging.h>
+#include <dsn/utility/flags.h>
+
+namespace dsn {
+namespace utils {
+
+DSN_DEFINE_bool("replication", enable_latency_tracer, true, "whether enable the latency tracer");
+
+latency_tracer::latency_tracer(int id, const std::string &name)
+    : name(fmt::format("(TRACE.{}.{})", name, id)), start_time(dsn_now_ns())
+{
+}
+
+void latency_tracer::add_point(const std::string &stage_name)
+{
+    if (!FLAGS_enable_latency_tracer) {
+        return;
+    }
+
+    int64_t ts = dsn_now_ns();
```

Comment:
the return type of `dsn_now_ns()` is `uint64_t`, you can use `uint64_t` in `points` also.